railties/lib/rails/generators/rails/encryption_key_file/encryption_key_file_generator.rb

```diff
@@ -44,13 +46,18 @@ def ignore_key_file(key_path, ignore: key_ignore(key_path))
         end
       end
 
-      def ignore_key_file_silently(key_path, ignore: key_ignore(key_path))
-        append_to_file ".gitignore", ignore if File.exist?(".gitignore")
+      def ensure_key_files_are_ignored_silently(key_path, ignore: key_ignore(key_path))
+        if File.exist?(".gitignore")
+          unless File.read(".gitignore").include?(ignore)
+            append_to_file ".gitignore", ignore
+          end
+        end
       end
 
       private
         def key_ignore(key_path)
-          [ "", "/#{key_path}", "" ].join("\n")
+          key_path = Pathname.new(key_path) unless key_path.is_a?(Pathname)
+          [ "", "# Ignore key files for decrypting credentials and more.", "/#{key_path.dirname.join("*.key")}", "" ].join("\n")
```

Comment:
```suggestion
          <<~IGNORE.squish
            
            # Ignore key files for decrypting credentials and more.
            /#{key_path.dirname.join("*.key")}
            
          IGNORE
```

You might want to inspect the output to make sure indents, etc line up.